lib/fs-watch/src/lib.rs

```diff
@@ -1,7 +1,10 @@
 #[macro_use]
 extern crate log;
+#[cfg(target_os = "linux")]
 #[macro_use]
 extern crate futures;
+#[cfg(not(target_os = "linux"))]
```

Comment:
I believe this can use `#[cfg_attr(target_os = "linux", macro_use)]`.

Alternatively, with newer Rust version, we can just use `futures::try_ready!` at the invocation point, I believe.